lib/githubbot.ts

```diff
@@ -251,7 +252,7 @@ export class GithubBot extends ProcBot.ProcBot<string> {
                         // If there's more retries, try again in 5 seconds.
                         setTimeout(runApi, 5000);
                     }
-                }).then(resolve);
+                });
             };
```

Comment:
Maybe this is overkill, but this block could be simpler with some TS's async/await + a little bluebird goodness. Something like:

```typescript
protected gitCall = async (method: any, options: any, retries?: number): Promise<any> => {
  let badCreds = false;
  let retriesLeft = retries || 3;

  while (true) {
    retriesLeft -= 1;
    try {
      return await method(options);
    } catch (err) {
      if ((err.message === 'Bad credentials') && !badCreds) {
        badCreds = true;
        await this.authenticate();
      } else if (retriesLeft === 0) {
        throw err;
      } else {
        await Promise.delay(5000);
      }
    }
  }  
}
```

Pretty sure this gives the same behaviour (including getting this fix automatically, since you don't have to `resolve` by hand), but imo reads much easier.